machine/v1/types_controlplanemachineset.go

```diff
@@ -250,6 +251,11 @@ type FailureDomains struct {
 	// +optional
 	GCP *[]GCPFailureDomain `json:"gcp,omitempty"`
 
+	// VSphere configures failure domain information for the VSphere platform.
```

Comment:
Nit, should be a lower case vsphere to match json casing, other fields here aren't correct
```suggestion
	// vsphere configures failure domain information for the VSphere platform.
```